civis/io/_tables.py

```diff
@@ -629,11 +641,133 @@ def csv_to_civis(filename, database, table, api_key=None, client=None,
     if archive:
         warnings.warn("`archive` is deprecated and will be removed in v2.0.0. "
                       "Use `hidden` instead.", FutureWarning)
+
+    name = path.basename(filename)
     with open(filename, "rb") as data:
-        fut = _import_bytes(data, database, table, client, max_errors,
-                            existing_table_rows, distkey, sortkey1, sortkey2,
-                            delimiter, headers, credential_id,
-                            polling_interval, archive, hidden=hidden)
+        file_id = file_to_civis(data, name, api_key=api_key, client=client)
+        fut = civisfile_to_civis(file_id, database, table, api_key=api_key,
+                                 client=client, max_errors=max_errors,
+                                 existing_table_rows=existing_table_rows,
+                                 distkey=distkey,
+                                 sortkey1=sortkey1, sortkey2=sortkey2,
+                                 delimiter=delimiter, headers=headers,
+                                 credential_id=credential_id,
+                                 polling_interval=polling_interval,
+                                 archive=archive, hidden=hidden)
+    return fut
+
+
+@deprecate_param('v2.0.0', 'api_key')
+def civisfile_to_civis(file_id, database, table, api_key=None, client=None,
```

Comment:
The name for this function is a little awkward, do you have any other ideas for what to call this?

Reply:
Here are some ideas: 

* `civis_file_to_table`,
* `civis_file_to_database`
* `remote_file_to_civis`

I can see how `civisfile_to_civis` follows the pattern we've been using, but I think it looks strange to have a function like `civis.io.civisfile_to_civis`...

Reply:
Yeah, totally agree. I was trying to stick with the current naming convention, but it is confusing. I like civis_file_to_table.